### PR TITLE
6861 - ET - Unregister onChange listeners

### PR DESCRIPTION
**Pull request by etelmon**
Jira Task: https://zimitio.atlassian.net/browse/ZWEB-6861

The onChanges listener was enhanced:
- It can now listen to changes emitted by the local server only instead of  both local ones AND those coming from the cluster.
- It can be unregistered. This helps with unit testing

**Steps to Reproduce**

Run:
`npm test`

**Performance Considerations**
No performance impact.

**Sections of the Application Affected**
Look up application pr to QA.

### CODE REVIEW

**Review by Semen-Lohunov — COMMENTED** [review]
Do not forget to update Jira task number to ZWEB-6861

File: lib/zerv-sync.js [hunk 1]
```diff
@@ -99,13 +98,15 @@ if (!sync) {
...
     };
-
     zervCore.addModule('Sync', sync);
+    sync._processExternalRecordNotications = _processExternalRecordNotications;
 
+    const cluster = clusterService.syncWithRedis(sync._processExternalRecordNotications, zervCore.transport);
 
-    module.exports = sync;
+    // add public functions to the core
+    zervCore.addModule('Sync', sync);
```
Comment (Semen-Lohunov) [inline lib/zerv-sync.js:109]: It duplicates line above.

File: lib/zerv-sync.js [hunk 2]
```diff
@@ -507,20 +517,22 @@ if (!sync) {
...
-                _.forEach(records, function(record) {
-                    // this make sure we show most silent error. If callback deals with promises it should return a promise
-                    try {
-                        const r = callback(tenantId, record, notificationType, options);
-                        if (r && r.then) {
-                            r.catch(function(err) {
-                                logger.error('Error in callback of the notification listener %b.', dataNotification, err.stack || err);
-                            });
+            _.forEach(listeners, (listener) => {  
+                if (options && (!options.clusterSrc || !listener.localOnly)) {
```
Comment (Semen-Lohunov) [inline lib/zerv-sync.js:521]: If options are not passed then it will not call listener?

File: lib/zerv-sync.js [hunk 3]
```diff
@@ -483,6 +488,11 @@ if (!sync) {
         return processRecordNotifications(tenantId, dataNotification, notificationType, objects, options);
     }
 
+    function _processExternalRecordNotications(tenantId, dataNotification, notificationType, objects, options) {
```
Comment (Semen-Lohunov) [inline lib/zerv-sync.js:491]: Typo "Notifications"

File: lib/zerv-sync.js (hunk 1 above)
Comment (etelmon) [reply lib/zerv-sync.js:109]: Removed

File: lib/zerv-sync.js (hunk 2 above)
Comment (etelmon) [reply lib/zerv-sync.js:521]: options is always provided. Not too clean. I changed the condition,

**Comment by etelmon** [comment]
This is depending on https://zimitio.atlassian.net/browse/ZWEB-6861. As soon as that 6861 gets to test, this code needs to be modified.

**Pull request merged**